### PR TITLE
Husky, ESLint, Lint-staged, mini-css-extract-plugin added

### PR DESCRIPTION
**Pull request by pmita**
Additional configuration for webpack-react project added. 
-Husky to automate git hooks
-ESLint to fix/anotate syntax problems
-Lint-stage to work in combination with husky
-Mini-css-extract-plugin to seperate our css from js during webpack bundling

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
+{
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended"
```
Comment (elkinuk) [inline .eslintrc.json]: let's add plugin:jsx-a11y/recommended too
can be a bit pain in the neck sometimes, but is disciplines really nicely

File: src/index.html [hunk 2]
```diff
@@ -4,7 +4,8 @@
     <meta charset="UTF-8">
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Document</title>
+    <link rel="stylesheet" href="main.css"/>
+    <title>Movie App</title>
```
Comment (elkinuk) [inline src/index.html]: you can add title with htmlWebpackPlugin to separate dev and prod

File: package.json [hunk 3]
```diff
@@ -5,24 +5,46 @@
...
+    "*.js": [
+      "npm run lint:fix",
+      "git add"
+    ]
   },
   "keywords": [],
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "husky": "^7.0.2",
```
Comment (elkinuk) [inline package.json]: husky is a dev dependency

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run lint-staged
```
Comment (elkinuk) [inline .husky/pre-commit]: 'npm run' starts script from 'package.json' with the name that you pass after it. It doesn't work
check https://www.npmjs.com/package/lint-staged installation paragraph

File: package.json [hunk 5]
```diff
@@ -5,24 +5,46 @@
...
     "test": "echo \"Error: no test specified\" && exit 1",
-    "serve" : "webpack serve --mode development",
-    "build" : "webpack --mode production"
+    "serve": "webpack serve --mode development",
+    "build": "webpack --mode production",
+    "lint": "eslint --ext .js,.jsx .",
+    "lint:fix": "eslint --fix --ext .js,.jsx .",
+    "prepare": "husky install"
+  },
+  "husky": {
```
Comment (elkinuk) [inline package.json]: this config is redundant for new versions of husky

**Review by elkinuk — CHANGES_REQUESTED** [review]
* Please, also add prettier package to maintain code style
now your App file doesn't have correct indentation
* You don't need both .eslintrc and .eslintrc.json
.eslintrc name is deprecated for some time now, so please, use .json
* Husky doesn't work properly. Try deleting all configuration and initing all again, using instruction here https://www.npmjs.com/package/husky. If you want to use lint-staged on pre-commit, after that go through installation in https://www.npmjs.com/package/lint-staged 
* Files, that won't be build should be stored in folder /public near /src. Files like fonts, some images, index.html will go there
* Any compilable files (styles, js) shouldn't be added in HTML template as tag. Use power of webpack to add them there. Browser is used to cash files, so in order to some changes to be applied we allow webpack to build HTML for us adding unique hash to the end of the files to prevent them from cashing
* Please, don't use ES6 imports in webpack config. There is no benefit in it, though some complications are added because of it
* Consider adding proper Readme.md to make startup process easier. It is really handy to describe all steps that should be done to start your aplication on ANY machine. Remember, every application is unique, same as every machine that we're working on. So it'll be beneficial for anyone who would like to start your app on their machine or for you in the future
* For the future, try using as less globally installed packages as possible, as it will complicate deploy process and running your app on the different machines. Command npx may come in hand

File: package.json [hunk 6]
```diff
@@ -5,24 +5,46 @@
...
+    "hooks": {
+      "pre-commit": "echo \"[Husky] pre-commit\""
+    }
+  },
+  "lint-staged": {
+    "*.js": [
+      "npm run lint:fix",
+      "git add"
```
Comment (elkinuk) [inline package.json]: using git add here might cause some changes that you're not intended to commit to be committed. Use just npm run lint instead

File: .husky/pre-commit [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+exit 1
```
Comment (elkinuk) [inline .husky/pre-commit]: Adding exit 1 in the end of this file will end every commit, even if all checks has passed
Witch brings the suspicion that your husky doesn't work

**Review by elkinuk — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (pmita) [reply package.json]: I removed lint-staged alltogether, no reason having it

**Pull request merged**